### PR TITLE
Block device from sleeping during certain activities.

### PR DESCRIPTION
**Pull request by charlesmchen**
PTAL @michaelkirk

### CODE REVIEW

**Comment by michaelkirk** [comment]
I haven't looked closely yet, but can we also prevent sleep when running DB migrations? Or would you prefer that be apart of another PR?

**Comment by charlesmchen** [comment]
PTAL @michaelkirk 

I pushed another commit: "Block device from sleeping while Debug UI is visible and during database upgrades."

File: Signal/src/util/DeviceSleepManager.swift [hunk 1]
```diff
@@ -0,0 +1,81 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import Foundation
+
+// This entity has responsibility for blocking the device from sleeping if
+// certain behaviors (e.g. recording or playing voice messages) are in progress.
+// 
+// Sleep blocking is keyed using "block objects" whose lifetime correponds to
```
Comment (michaelkirk) [inline Signal/src/util/DeviceSleepManager.swift]: > // Sleep blocking is keyed using "block objects" whose lifetime correponds to
> // Sleep blocking is keyed using "block objects" whose lifetime corre__s__ponds to

Comment (charlesmchen) [inline Signal/src/util/DeviceSleepManager.swift]: Done.

**Pull request merged**